File: Getting Started - WCF RIA Services/[C#]-Getting Started - WCF RIA Services/C#/HRApp/Helpers/NotOperatorValueConverter.cs
namespace HRApp
{
    using System;
    using System.Windows.Data;

    /// <summary>
    ///     Two way IValueConverter that lets you bind the inverse of a boolean property
    ///     to a dependency property
    /// </summary>
    public class NotOperatorValueConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture)
        {
            return !((bool)value);
        }

        public object ConvertBack(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture)
        {
            return !((bool)value);
        }
    }
}

File: Getting Started - WCF RIA Services/[VB]-Getting Started - WCF RIA Services/VB/HRApp/Helpers/NotOperatorValueConverter.vb
Imports System
Imports System.Windows.Data

''' <summary>
'''     Two way IValueConverter that lets you bind the inverse of a boolean property
'''     to a dependency property
''' </summary>
Public Class NotOperatorValueConverter
    Implements IValueConverter


    Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IValueConverter.Convert
        Return Not DirectCast(value, Boolean)
    End Function

    Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IValueConverter.ConvertBack
        Return Not DirectCast(value, Boolean)
    End Function
End Class